host/xtest/xtest_6000.c

```diff
@@ -1090,143 +1150,191 @@ static void xtest_tee_test_6009_single(ADBG_Case_t *c, uint32_t storage_id)
 
 DEFINE_TEST_MULTIPLE_STORAGE_IDS(xtest_tee_test_6009)
 
-#ifdef WITH_GP_TESTS
-static void xtest_tee_test_6010(ADBG_Case_t *c)
+static void xtest_tee_test_6010_single(ADBG_Case_t *c, uint32_t storage_id)
 {
-    TEEC_Session sess;
-    uint32_t orig;
-    uint8_t out[4000] = {0};
-    uint8_t in[0x12c] = {'b'};
-    int i;
+	TEEC_Session sess;
+	uint32_t orig;
+	uint32_t o1;
+	uint32_t o2;
+	uint32_t e;
+	uint32_t f;
+	uint8_t data[1024];
+	uint8_t out[1024];
+	uint32_t n;
 
-    for (i=0; i<sizeof(in); i++)
-	    in[i] = i;
+	for (n = 0; n < ARRAY_SIZE(data); n++)
+		data[n] = n;
 
-    if (!ADBG_EXPECT_TEEC_SUCCESS(
-            c, xtest_teec_open_session(
-                &sess, &gp_tta_ds_uuid, NULL, &orig)))
-        return;
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&sess, &storage_ta_uuid, NULL, &orig)))
+		return;
 
-    Do_ADBG_BeginSubCase(
-        c, "GP DS CreatePersistentObject AccessConflict (9d-1d-62)");
+	Do_ADBG_BeginSubCase(c, "CreatePersistentObject AccessConflict");
 
-    if (!ADBG_EXPECT_TEEC_RESULT(
-            c, TEEC_ERROR_ACCESS_CONFLICT, ds_open_access_conf(&sess)))
-        goto exit;
+	o1 = TEE_HANDLE_NULL;
+	o2 = TEE_HANDLE_NULL;
+	f = TEE_DATA_FLAG_ACCESS_READ | TEE_DATA_FLAG_ACCESS_WRITE |
+	    TEE_DATA_FLAG_ACCESS_WRITE_META | TEE_DATA_FLAG_SHARE_READ |
+	    TEE_DATA_FLAG_SHARE_WRITE | TEE_DATA_FLAG_OVERWRITE;
 
-    Do_ADBG_EndSubCase(
-        c, "GP DS CreatePersistentObject AccessConflict (9d-1d-62)");
-    Do_ADBG_BeginSubCase(
-        c, "GP DS RestrictObjectUsagePanic (9d-5d-46)");
+	ADBG_EXPECT_TEEC_SUCCESS(c,
+		fs_create(&sess, file_00, sizeof(file_00), f, 0, data,
+			  sizeof(data), &o1, storage_id));
 
-    if (!ADBG_EXPECT_TEEC_RESULT(
-            c, TEE_ERROR_TARGET_DEAD, ds_res_obj_panic(&sess)))
-        goto exit;
+	f &= ~TEE_DATA_FLAG_OVERWRITE;
```

Comment:
The GP TA sets the flags to `TEE_DATA_FLAG_ACCESS_READ | TEE_DATA_FLAG_ACCESS_WRITE`, it does not just clear `TEE_DATA_FLAG_OVERWRITE`.